proksee/species_estimator.py

```diff
@@ -161,27 +161,42 @@ def run_mash(self):
 
         LINE_LENGTH_LIMIT = 3500  # Actually 4095, but smaller here for safety.
 
-        output_filepath = os.path.join(self.output_directory, self.OUTPUT_FILENAME)
+        output_filepath = os.path.abspath(os.path.join(self.output_directory, self.OUTPUT_FILENAME))
+
+        # Find the common directory between all paths:
+        # This is a safer solution than assuming everything will always use the output directory.
+        common_path = os.path.commonpath(self.input_list)
+
+        if os.path.isdir(common_path):
+            common_directory = common_path
+        else:
+            # The common path is NOT a directory.
+            # This can happen when there is only a single input (the common path is the single file's filepath).
+            common_directory = os.path.dirname(common_path)
 
         # create the mash command
+        # Use the full file path for the database file:
         command = "mash screen -i 0 -v 1 " + self.mash_database_filename
 
         for item in self.input_list:
-            command += " " + str(item)
+            # Grab the relative path from the common directory of each item:
+            command += " " + str(os.path.relpath(item, start=common_directory))
```

Comment:
Do you know if `os.path.relpath` or other commands above will properly escape spaces and other special characters in path names?

Reply:
I've converted over the command format from being a string to a list of arguments. Some other changes happened to make this work:

- File piping was replaced with an intermediate file, partly to be safer with potentially large mash outputs.
- The shell is no longer used.
- Checking of the command line length is now done a bit different. It checks the length of the relative paths. In practice this MIGHT actually be closer to 64000, rather than the 3500 we use, but it might depend on implementations, so I went with the smaller number.